tracks/go/exercises/raindrops/README.md

```diff
@@ -0,0 +1,65 @@
+# Raindrops
+
+This is the third core exercise in the go track to receive feedback from a mentor. It is less about learning new concepts but more about deepening the already acquired knowledge. To some the conversion of an `int` to `string` and the modulus operator `%` are new.
+
+This may be an excellent place to talk about the go tooling again -- especially if the tools were not applied (`gofmt`, `golint`, etc).
+
+## Reasonable solutions
+
+The exercise can be solved with `if` statements and `string` concatenation:
+
+
+```
+// Convert implements raindrop speech
+func Convert(i int) string {
+	var res string
+	if i%3 == 0 {
+		res += "Pling"
+	}
+	if i%5 == 0 {
+		res += "Plang"
+	}
+	if i%7 == 0 {
+		res += "Plong"
+	}
+
+	if res == "" {
+		res = strconv.Itoa(i)
+	}
+
+	return res
+}
+```
+
+## Common suggestions
+
+The most common feedback revolves around:
+
+* General programming things (working with strings, map iteration)
+* Readability (formatting, comments, linting)
+
+**Map: random iteration** 
+* are they defining static values (map, slice) in the function? Suggest moving them to package level.
+* are they using a `map` to hold the cases? This gets complicated as they need to somehow make sure the order is correct. From the [specs](https://golang.org/ref/spec#For_statements): "The iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next." You could direct them towards using `for i := 3; i < 8; i += 2` or a for loop that iterates the slice `[]int{3, 5, 7}`
+
+**Control flows**
+* are they using `if { return ..} else { return ..}`? Point them to `golint` which will suggest to remove the else and outdent its block.
+* are they using a `for i; .. i++` loop? Check what range they are iterating. The most optimal loop iterates `for i := 3; i < 8; i += 2`. Iterating to given `number` (as many do) is not necessary.
+
+**Strings: concatenation and conversion**
+* are they using `strconv.FormatInt`? You can point them to `strconv.Itoa` as being more convenient (no `int64` conversion needed, no second parameter).
+* are they using a slice and `strings.Join`? You could mention appending to the string with `+=` as a simpler method.
+* they are using a `strings.Builder`? You could mention appending to the string with `+=` as a simpler method or -- if they seem to be more advanced or are very hungry -- as an option that consumes ~2.5 times less memory.
+
+**Inconsistent formatting: mention `gofmt`**
+* Did they not run `gofmt`? This is harder to detect right now unless it's blatantly badly formatted, but if it's clear that `gofmt` hasn't been run, it's worth gently pointing to it (https://blog.golang.org/go-fmt-your-code), and maybe mentioning that this is the accepted standard, and that all professional or open source projects that you contribute to will require you to run it.
```

Comment:
By now, if you spot that they haven't run `gofmt`, require that they run it before approving!

Reply:
ok. Will add that. Mentioned that to be enforced in the next (Scrabble) but probably a good idea to enforce gofmt here and golint from the next onward.